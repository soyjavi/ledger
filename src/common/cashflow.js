import C from './constants';

const { TX: { TYPE } } = C;

export default (txs = []) => {
  let income = 0;
  let expenses = 0;

  txs.forEach(({ type, value }) => {
    if (type === TYPE.EXPENSE) expenses += value;
    else if (type === TYPE.INCOME) income += value;
  });

  return { income, expenses };
};
