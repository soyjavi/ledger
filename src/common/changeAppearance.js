import { Theme } from '@lookiero/aurora';

import { ShieldTheme } from '@theming';

export const changeAppearance = async ({ color, updateSettings }) => {
  Theme.set({ ...ShieldTheme, $theme: `shield-${color.toString()}`, $colorPrimary: color });

  if (updateSettings) await updateSettings({ appearance: color });
};
