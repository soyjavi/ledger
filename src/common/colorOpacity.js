import color from 'color';

export const colorOpacity = (value, opacity = 0.5) => color(value).alpha(opacity).string();
