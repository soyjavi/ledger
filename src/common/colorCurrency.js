import { THEME } from 'reactor/common';

import { C } from './constants';

const { CURRENCY_COLOR, CURRENCY } = C;
const { COLOR } = THEME;

export const colorCurrency = (currency = CURRENCY) => CURRENCY_COLOR[currency] || COLOR.TEXT;
