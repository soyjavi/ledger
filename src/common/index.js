import C from './constants';
import cashflow from './cashflow';
import exchange from './exchange';
import getLocationAsync from './getLocationAsync';
import FORM from './forms';
import L10N from './l10n';
import fetch from './fetch';
import theme from './theme';
import translate from './translate';
import verboseDate from './verboseDate';

export {
  C,
  cashflow,
  exchange,
  getLocationAsync,
  FORM,
  L10N,
  fetch,
  theme,
  translate,
  verboseDate,
};
