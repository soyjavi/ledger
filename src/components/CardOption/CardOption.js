import { MaterialCommunityIcons } from '@expo/vector-icons';
import {
  bool, func, node, number, oneOfType, string,
} from 'prop-types';
import React from 'react';
import { Image, View } from 'react-native';

import { Text, Touchable } from '../../reactor/components';
import { THEME } from '../../reactor/common';
import { Box } from '../Box';
import styles from './CardOption.style';

const { COLOR } = THEME;

const CardOption = ({
  children, color, icon, image, onPress, selected, title, ...inherit
}) => (
  <Box small style={[styles.box, inherit.style]}>
    <Touchable
      onPress={onPress}
      rippleColor={COLOR.PRIMARY}
      style={[styles.container, selected && { backgroundColor: color }]}
    >
      { icon && (
        <View style={styles.icon}>
          <MaterialCommunityIcons name={icon} color={COLOR.TEXT} size={28} />
        </View>
      )}

      { image && <Image source={image} style={styles.image} /> }

      <Text caption lighten={color === COLOR.PRIMARY} numberOfLines={1} style={styles.title}>
        {title}
      </Text>
      { children }
    </Touchable>
  </Box>
);

CardOption.propTypes = {
  children: node,
  color: string,
  icon: oneOfType([number, string]),
  image: oneOfType([number, string]),
  onPress: func.isRequired,
  selected: bool,
  title: string.isRequired,
};

CardOption.defaultProps = {
  children: undefined,
  color: COLOR.PRIMARY,
  icon: undefined,
  image: undefined,
  selected: false,
};

export { CardOption };
