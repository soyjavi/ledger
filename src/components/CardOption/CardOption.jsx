import { bool, func, node, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { Image, View } from 'react-native';

import { Icon, Text, Touchable } from '../../reactor/components';
import { THEME } from '../../reactor/common';
import { Box } from '../Box';
import styles from './CardOption.style';

const { COLOR } = THEME;

const CardOption = ({ children, color, icon, image, onPress, selected, title, ...inherit }) => {
  let colorContent = selected ? COLOR.BACKGROUND : COLOR.TEXT_LIGHTEN;

  return (
    <Box small style={[styles.box, inherit.style]}>
      <Touchable
        onPress={onPress}
        rippleColor={COLOR.TEXT}
        style={[styles.container, selected && { backgroundColor: color }]}
      >
        {icon && (
          <View style={styles.icon}>
            <Icon value={icon} color={colorContent} size={28} />
          </View>
        )}

        {image && <Image source={image} style={styles.image} />}

        <Text caption color={colorContent} numberOfLines={1} style={styles.title}>
          {title}
        </Text>
        {children}
      </Touchable>
    </Box>
  );
};

CardOption.propTypes = {
  children: node,
  color: string,
  icon: oneOfType([number, string]),
  image: oneOfType([number, string]),
  onPress: func.isRequired,
  selected: bool,
  title: string.isRequired,
};

CardOption.defaultProps = {
  children: undefined,
  color: COLOR.TEXT,
  icon: undefined,
  image: undefined,
  selected: false,
};

export { CardOption };
