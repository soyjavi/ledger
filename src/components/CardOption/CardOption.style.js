import { StyleSheet } from 'react-native';

import { THEME } from '../../reactor/common';

const { BORDER_RADIUS, FONT, SPACE, UNIT } = THEME;
const IMAGE_SIZE = SPACE.L;

const CARD_OPTION_WIDTH = UNIT * 8;

export default StyleSheet.create({
  box: {
    width: CARD_OPTION_WIDTH,
  },

  container: {
    alignItems: 'center',
    alignContent: 'center',
    justifyContent: 'center',
    borderRadius: BORDER_RADIUS,
    paddingHorizontal: SPACE.XS,
    height: '100%',
    width: '100%',
  },

  image: {
    height: IMAGE_SIZE,
    width: IMAGE_SIZE,
  },

  legend: FONT.LEGEND,
});
