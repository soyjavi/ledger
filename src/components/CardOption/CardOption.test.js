import React from 'react';
import renderer from 'react-test-renderer';

import { C } from '../../common';
import CardOption from './CardOption';

describe('<CardOption>', () => {
  it('renders', () => {
    const tree = renderer.create(<CardOption />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('when {...}', () => {
  // });
});

