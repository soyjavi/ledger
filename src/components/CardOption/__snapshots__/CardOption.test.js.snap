// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardOption> renders 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4281545523,
        4280295456,
        4280295456,
      ],
      "endPoint": Array [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0,
        0,
      ],
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#cdd2d4",
        "borderRadius": 3,
        "height": "100%",
        "overflow": "hidden",
        "padding": 16,
        "width": "100%",
      },
      Array [
        Object {
          "marginRight": 12,
          "padding": 0,
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="none"
    style={
      Array [
        undefined,
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "padding": 16,
          },
          false,
        ],
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#10c0bc",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
